app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -548,23 +548,22 @@ BrowserWorld::State::UpdateControllers(bool& aRelayoutWidgets) {
     if (controller.modelToggle)
       controller.modelToggle->ToggleAll(controller.mode == ControllerMode::Device);
 
+    // If we are rendering hands using spheres, update the joints' transforms here.
+    if (controller.handJointTransforms.size() > 0) {
+      assert(controller.handJointTransforms.size() == controller.meshJointTransforms.size());
+      for (int i = 0; i < controller.handJointTransforms.size(); i++)
+        controller.handJointTransforms[i]->SetTransform(controller.meshJointTransforms[i]);
```

Comment:
I don't understand why we are doing this ? If we are rendering the spheres, like the comment above suggest, we are using the mesh's transforms for the handJoint transforms ?

Reply:
Note that this is the same we were doing in ControllerContainer::SetHandJointLocations(), but now we just stored the matrices there, and we assign them to the vrb::Transform of the spheres here. It is the same logic as before, but split in two and moved to a different part of the code.

Reply:
But why we can't simply use the controller.meshJointTransforms instead copying them to the handJointTransforms vector ? This is also related to my previous question about why we need 2 different vectors.

Reply:
Because `meshJointTransforms` is just a vector of vrb::Matrix. We need a vrb::Transform node to position the spheres in the correct location and orientation.